Islandora-CLAW/islandora
# Issue 526

**GitHub Issue**: Part of Islandora-CLAW/CLAW#526

* Other Relevant Links:
  * https://github.com/Islandora-CLAW/islandora_collection/pull/14
  * Islandora-CLAW/CLAW#549

# What does this Pull Request do?

Pulls out the tracking of entity updates into its own service.

# What's new?
The number of times an entity has been updated is no longer stored as a field.  A service acts upon the entity during the core entity CRUD hooks. Because it's not a field, this removes it from having to be exposed in the ontology.  

I also rename vclock to version counter all over the place.

* Does this change require documentation to be updated? No
* Does this change add any new dependencies? No
* Does this change require any other modifications to be made to the repository
 (ie. Regeneration activity, etc.)? You'll need to uninstall and re-install the module, since there's a hook_schema implementation in the .install file.

# How should this be tested?

The kernel tests should take care of it, but you can manually run through the same steps.  By running `$ drupal database:client`, you can obtain a connection to the drupal database.  You can use the query `SELECT * from islandora_version_count;` to inspect the version count table between each step.
* Create an entity through the UI (this will initialize its row in the db to 0)
* Update the entity through the UI (this will increment its row in the db to 1)
* Delete the entity through the UI (this will delete its row in the db)

Also, if you request jsonld for a RdfSource via REST, you shouldn't see an entry for `islandora:vclock`.
 
# Interested parties
@Islandora-CLAW/committers
